### PR TITLE
docs: clarify RealtimeMetricsMonitor, HistoricalMetricsMonitor

### PR DESCRIPTION
**Pull request by techdocsmith**
Removes the obsolete `RealtimeMetricsMonitor`, clarifies that its metrics are now part of ingestion tasks. Clarifies you should enable `HistoricalMetricsMonitor` on Historicals.

I've noted there are some more references to realtime in the config topic, but it's not clear if those should all be moved, or, if like the metrics are now handled by other processes.

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -362,12 +362,11 @@ The following monitors are available:
 |----|-----------|
 |`org.apache.druid.client.cache.CacheMonitor`|Emits metrics (to logs) about the segment results cache for Historical and Broker processes. Reports typical cache statistics include hits, misses, rates, and size (bytes and number of entries), as well as timeouts and and errors.|
 |`org.apache.druid.java.util.metrics.SysMonitor`|Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using `SysMonitor`.|
-|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes.|
+|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes. Must be enabled on Historical processs.|
```
Comment (techdocsmith) [inline docs/configuration/index.md:365]:
```suggestion
|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes. Must be enabled on Historical process.|
```

File: docs/configuration/index.md [hunk 2]
```diff
@@ -362,12 +362,11 @@ The following monitors are available:
 |----|-----------|
 |`org.apache.druid.client.cache.CacheMonitor`|Emits metrics (to logs) about the segment results cache for Historical and Broker processes. Reports typical cache statistics include hits, misses, rates, and size (bytes and number of entries), as well as timeouts and and errors.|
 |`org.apache.druid.java.util.metrics.SysMonitor`|Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using `SysMonitor`.|
-|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes.|
+|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes. Must be enabled on Historical process.|
```
Comment (sthetland) [inline docs/configuration/index.md:365]: Nit, but I'd reword this to say that this can be enabled only on historical processes, which conveys both that it must be enabled there and trying to do so on other processes will produce errors. 

```suggestion
|`org.apache.druid.server.metrics.HistoricalMetricsMonitor`|Reports statistics on Historical processes. Available only on Historical processes.|
```

**Review by sthetland — CHANGES_REQUESTED** [review]

**Review by sthetland — APPROVED** [review]
LGTM. 

(The build check failed, but it's unrelated. The docs check passed.)
